pkg/netpol/connlist/exposure_map.go

```diff
@@ -35,91 +36,59 @@ func (ex exposureMap) addPeerGeneralExposure(pe *eval.PolicyEngine, peer Peer, i
 
 // addPeerUnprotectedData getting a peer and a direction; checks if the peer is not protected on that direction;
 // if not protected, i.e. not selected by any network-policy in the manifests in the given direction:
-// adds the peer to the exposure map and returns an indication that an entry was added to the map
-func (ex exposureMap) addPeerUnprotectedData(pe *eval.PolicyEngine, peer Peer, isIngress bool) (bool, error) {
+// adds the peer to the xgress exposure map and returns an indication that an entry was added to the map
+func (ex *exposureMaps) addPeerUnprotectedData(pe *eval.PolicyEngine, peer Peer, isIngress bool) (bool, error) {
 	isProtected, err := pe.IsPeerProtected(peer, isIngress)
 	if err != nil {
 		return false, err
 	}
-	_, ok := ex[peer]
-	// if the peer is not protected by any policy in the given direction; we want to add an entry to the map
-	if !ok && !isProtected {
-		if isIngress {
-			ex.addNewEntry(peer, notProtected, unknown)
-		} else {
-			ex.addNewEntry(peer, unknown, notProtected)
-		}
+	if !isProtected {
+		ex.addNewEntry(peer, isProtected, isIngress)
 		return true, nil
 	}
-	// else :if an entry exists; update the value in the map according to the result (protected/not protected)
-	if ok {
-		ex.updatePeerEntryWithCorrectProtectedData(peer, isIngress, isProtected)
-	}
 	return false, nil
 }
 
-func (ex exposureMap) updatePeerEntryWithCorrectProtectedData(peer Peer, isIngress, isProtected bool) {
-	if isIngress {
-		switch isProtected {
-		case true:
-			ex[peer].isIngressProtected = protected
-		case false:
-			ex[peer].isIngressProtected = notProtected
-		}
-	} else { // egress
-		switch isProtected {
-		case true:
-			ex[peer].isEgressProtected = protected
-		case false:
-			ex[peer].isEgressProtected = notProtected
-		}
-	}
-}
-
 // addPeerXgressEntireClusterExp checks and adds (if exists) ingress/egress entire cluster exposure for the given peer
 // on the given direction
-func (ex exposureMap) addPeerXgressEntireClusterExp(pe *eval.PolicyEngine, peer Peer, isIngress bool) error {
+func (ex *exposureMaps) addPeerXgressEntireClusterExp(pe *eval.PolicyEngine, peer Peer, isIngress bool) error {
 	conn, err := pe.GetPeerXgressEntireClusterConn(peer, isIngress)
 	if err != nil {
 		return err
 	}
 	if conn.IsEmpty() {
 		return nil
 	}
-	// if there is no map entry for the peer : add new entry
-	// exposed to entire cluster: means that there is at least one network-policy exposing the peer to entire cluster
-	if _, ok := ex[peer]; !ok {
-		if isIngress {
-			ex.addNewEntry(peer, protected, unknown)
-		} else {
-			ex.addNewEntry(peer, unknown, protected)
-		}
-	}
-	// update the entry of the peer with the entire cluster connection
+	// exposed to entire cluster - first entry of the peer
+	ex.addNewEntry(peer, true, isIngress)
```

Comment:
why is it certain that the peer is not in the map?

Reply:
because for each peer we check and update in the relevant map its "general conns" i.e. if unprotected, or there is connection with entire cluster; **only once** and before  adding any other connection with a representative peer.
so since, we have now separate map for each direction , we are for sure know there is no previos entry in the map with this peer. (that was the cause of the "older" fixes in this pr).

however, if I apply [this](https://github.com/np-guard/netpol-analyzer/pull/316#discussion_r1546241978); 
it will always check existence